### PR TITLE
Osiris/e2e tests

### PR DESCRIPTION
**Pull request by 0xOsiris**
Tests Block building, and priority ordering e2e through the `WorldChainEthApi`

### CODE REVIEW

File: world-chain-builder/src/e2e_tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        )?)
+        .launch()
+        .await?;
+
+    let wallets = Wallet::new(10).with_chain_id(chain_id).gen();
+
+    return Ok((NodeTestContext::new(node).await?, wallets, chain_id, tasks));
+}
+
+type Adapter = NodeAdapter<
```
Comment (Dzejkop) [inline world-chain-builder/src/e2e_tests/mod.rs:60]: wow

Comment (Dzejkop) [reply world-chain-builder/src/e2e_tests/mod.rs:60]: Can we move this and NodeTestContextType to the top of the file? Right below imports?

File: world-chain-builder/src/e2e_tests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    >,
+>;
+
+pub type NodeTestContextType = NodeTestContext<Adapter, WorldChainAddOns>;
+
+#[tokio::test]
+async fn test_can_build_mixed_pbh_payload() -> eyre::Result<()> {
+    reth_tracing::init_test_tracing();
+    // Create a raw signed transfer
+    let raw_tx = TransactionTestContext::transfer_tx_bytes(8453, DEV_SIGNER.parse().unwrap()).await;
```
Comment (Dzejkop) [inline world-chain-builder/src/e2e_tests/mod.rs]: Can we extract 8453 to a const? And also use it in the second call to `transfer_tx_bytes`?

File: world-chain-builder/src/e2e_tests/mod.rs [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+    // Re-encode the envolope
+    let mut buff = Vec::<u8>::new();
+    world_chain_pooled_tx_element.encode_enveloped(&mut buff);
+
+    // Pre-validate the decoding
+    let pooled_transaction_element = recover_raw_transaction(buff.clone().into());
+    assert!(
+        pooled_transaction_element.is_ok_and(|e| e.0.semaphore_proof.is_some_and(|p| p == proof))
+    );
```
Comment (Dzejkop) [inline world-chain-builder/src/e2e_tests/mod.rs]: We could buble the error here and assert directly on the value, no?

Comment (Dzejkop) [reply world-chain-builder/src/e2e_tests/mod.rs]: Also this is just checking our encoding/decoding? Feels a bit redundant

File: world-chain-builder/src/e2e_tests/mod.rs [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+    let (mut node, mut dev_wallets, id, _tasks) = setup(chain_spec).await?;
+    let std_wallet = dev_wallets.pop().unwrap();
+    // inject normal tx
+    let raw_tx = TransactionTestContext::transfer_tx_bytes(id, std_wallet.clone()).await;
+    // Call eth_sendRawTransaction with the enveloped semaphore tx.
+    let semaphore_res = node.rpc.inject_tx(buff.into()).await;
+    let res = node.rpc.inject_tx(raw_tx).await;
+
+    assert!(res.is_ok());
+    assert!(semaphore_res.is_ok());
```
Comment (Dzejkop) [inline world-chain-builder/src/e2e_tests/mod.rs]: Let's just buble the errors with `?`, there's really no difference, and bubbling gives better error visiblity

File: world-chain-builder/src/e2e_tests/mod.rs (hunk 1 above)
Comment (0xForerunner) [reply world-chain-builder/src/e2e_tests/mod.rs:60]: Damn that is a serious type

File: world-chain-builder/src/e2e_tests/mod.rs (hunk 3 above)
Comment (0xOsiris) [reply world-chain-builder/src/e2e_tests/mod.rs]: Yeah I added this to isolate a bug early on. Will remove!

**Pull request merged**